microsoft-edge/webview2/how-to/hostobject.md

```diff
@@ -0,0 +1,262 @@
+    27
+    28            // Creates a date object on the native side and sets the DateProperty to it.
+    29            HRESULT CreateNativeDate();
+    30
+    31        };
+    32        //! [AddHostObjectInterface]
+    33
+    34        [uuid(637abc45-11f7-4dde-84b4-317d62a638d3)]
+    35        coclass HostObjectSample
+    36        {
+    37            [default] interface IHostObjectSample;
+    38            interface IDispatch;
+    39        };
+    40    }
+    ```
+
+1. On line 38, we include `interface IDispatch`, which is needed for our host object to work with `AddHostObjectToScript`.
+
+    **About IDispatch**:
+
+    `IDispatch` allows you to dynamically invoke methods and properties. Normally, calling objects requires static invocations, but you can use JavaScript to dynamically create object calls. For more information about `IDispatch` inheritance and methods, see [IDispatch interface (oaidl.h)](/windows/win32/api/oaidl/nn-oaidl-idispatch). 
+    
+    Implement `IDispatch` as described in [Type Libraries and the Object Description Language](/previous-versions/windows/desktop/automat/type-libraries-and-the-object-description-language).
+    
+    If the object you want to add to JavaScript doesn't already implement `IDispatch`, you need to write an `IDispatch` class wrapper for the object that you want to expose.
+
+    There might be libraries to do this automatically.  To learn more about the steps that are needed to write an `IDispatch` class wrapper for the object that you want to expose, see [Automation](/previous-versions/windows/desktop/automat/automation-programming-reference).
+    
+1. Once the interface is defined in the IDL, save and compile the sample project in Visual Studio to create the translation lookaside buffer (TLB) file. You need to reference the TLB file from the C++ source code shown in the following section.
+
+### Part 2B: Create the C++ object
+
+In the WebView2 sample code, the `HostObjectSampleImpl.cpp` file takes the skeleton created in the COM IDL file and builds the C++ object.
+
+Implement all the functions that are defined in your object's interface, as we outlined in the IDL file. Be sure to implement the functions that are required by `IDispatch`.  The compiler will throw a warning if these functions aren't defined.
+
+Next, we examine two specific properties that were defined in the IDL, to show how the IDL is related to the `.cpp` file.
+
+1. In Visual Studio **Solution Explorer**, open **WebView2APISample** > **Source Files** > **HostObjectSampleImpl.cpp**.
+
+1. Compare the property *declarations*, in `HostObjectSample.idl` ...
+ 
+    ```csharp
+    [propget] HRESULT Property([out, retval] BSTR* stringResult);
+    [propput] HRESULT Property([in] BSTR stringValue);
+    ```
+    ... to the *implementation* of the object's properties, in `HostObjectSampleImpl.cpp`:
+
+    ```cpp
+    STDMETHODIMP HostObjectSample::get_Property(BSTR* stringResult)
```

Comment:
@jasonstephen15 do we have reference as to why `[propget] HRESULT Property` is `get_Property` in implementation?